Add grid layout. (#881)
telerik/maui-docs: controls/collectionview/layouts/grid-layout.md

---
title: Grid Layout
page_title: .NET MAUI CollectionView Documentation - Grid Layout

> description: Learn how to configure the CollectionView to display the items vertically or horizontally.

Reviewer: ```suggestion
description: Learn how to configure the CollectionView to display its items in a grid-like layout and arrange them vertically or horizontally.
```

position: 1
slug: collectionview-grid-layout
tags: .net maui, collectionview, layout, grid layout
---

# .NET MAUI CollectionView Grid Layout

> The CollectionView control supports grid layout through its `ItemsLayout` property. The `ItemsLayout` accepts values of type `CollectionViewLayoutBase` which is a base class.

Reviewer: ```suggestion
The CollectionView control provides a grid-like layout that you can specify by using the `ItemsLayout` property. The `ItemsLayout` property accepts values of type `CollectionViewLayoutBase`, which is a base class.
```

> The grid layout allows distributing cells in a fixed number of columns/rows. The grid layout can be set by creating an instance of the `CollectionViewGridLayout` class and assigning it to the `RadCollectionView.ItemsLayout` property.

Reviewer: ```suggestion
The grid layout allows distributing cells in a fixed number of columns and rows. To set the grid layout, create an instance of the `CollectionViewGridLayout` class and assign it to the `RadCollectionView.ItemsLayout` property.
```

> Sample `RadCollectionView` definition with `CollectionViewGridLayout`:

Reviewer: Use complete sentences.
```suggestion
The following code snippet represents a sample `RadCollectionView` definition with `CollectionViewGridLayout`:
```

```XAML
<telerik:RadCollectionView>
    <telerik:RadCollectionView.ItemsLayout>
        <telerik:CollectionViewGridLayout />
    </telerik:RadCollectionView.ItemsLayout>
</telerik:RadCollectionView>
```

> This is how the items are arranged when grid layout is defined:

Reviewer: Use active instead of passive voice.
```suggestion
This is how the items are arranged when you define a grid layout:
```

![.NET MAUI CollectionView Grid Layout](../images/collectionview-grid-layout.png "Telerik .NET MAUI CollectionView")

> The properties available in the `CollectionViewGridLayout` are: 

Reviewer: ```suggestion
The `CollectionViewGridLayout` provides the following properties:
```

> * `SpanCount` (`int`)&mdash;Specifies the count of the columns or rows (depending on the orientation) of the layout.The value must be `>= 1`. The default value is `2`.

Reviewer: ```suggestion
* `SpanCount` (`int`)&mdash;Specifies the count of the columns or rows (depending on the orientation) of the layout. The value must be `>= 1`. The default value is `2`.
```

* `HorizontalItemSpacing` (`double`)&mdash;Specifies the horizontal empty space around each item. The default value is `0.0`.

> * `VerticalItemSpacing` (`double`)&mdash;Specifies the vertical empty space around each item.The default value is `0.0`.

Reviewer: ```suggestion
* `VerticalItemSpacing` (`double`)&mdash;Specifies the vertical empty space around each item. The default value is `0.0`.
```

> * `Orientation` (enum or type `Telerik.Maui.Orientation`)&mdash;Specifies the orientation of the layout. The available options are: `Horizontal` and (default) `Vertical`.

Reviewer: ```suggestion
* `Orientation` (enum or type `Telerik.Maui.Orientation`)&mdash;Specifies the orientation of the layout. The available options are: `Horizontal` and `Vertical` (default).
```

## Example with Grid Layout

> The following example demonstrates how to apply grid layout to the `RadCollectionView`.

Reviewer: ```suggestion
The following example demonstrates how to apply the grid layout to the `RadCollectionView`.
```

 
**1.** Define the following business object:

<snippet id='collectionview-datamodel' />

**2.** Create a `ViewModel` class as shown below:

<snippet id='collectionview-viewmodel' />

> **3.** Add the `RadCollectionView` definition with grid layout:

Reviewer: ```suggestion
**3.** Add the `RadCollectionView` definition with the grid layout:
```
